core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -13,6 +13,8 @@
  */
 package io.trino.spi.connector;
 
+import io.trino.spi.predicate.TupleDomain;
+
 import java.util.List;
 
 public interface ConnectorPageSourceProvider
```

Comment:
> and not RecordSetProvider

`RecordSetProviders` usually don't have granular splits compared to lakehouse connectors and they don't do opportunistic filtering, hence the methods here are less useful for records. `ConnectorRecordSetProvider` also doesn't accept dynamic filter as argument. For `RecordSetProviders` these methods could probably be NOOPs

> as they have nothing to do with a data stream itself. They are more related to split management.

Actually it's quite related. Lakehouse connectors use `getUnenforcedPredicate` (which in turn uses `prunePredicate`) when creating page source. They've always done that albeit it was not formalized. Now it's formalized and exposed to engine as additional per-split metainfo. It's important for correctness too since cache needs to know what opportunistic predicate was used to filter stream data.

I was thinking about something like `ConnectorSplitInfoProvider`, but it would have to be hooked to `PageSourceProviders` internally anyway (and probably rooted at `ConnectorPageSourceProviderFactory`). Hence, I think the current location of these methods is probably optimal.